mining/mining.go

```diff
@@ -620,14 +627,38 @@ mempoolLoop:
 		switch {
 		// If segregated witness has not been activated yet, then we
 		// shouldn't include any witness transactions in the block.
-		case tx.MsgTx().HasWitness() && !includeWitness:
+		case tx.MsgTx().HasWitness() && !segwitActive:
 			continue
 
 		// Otherwise, Keep track of if we've included a transaction
 		// with witness data or not. If so, then we'll need to include
 		// the witness commitment as the last output in the coinbase
 		// transaction.
-		case tx.MsgTx().HasWitness() && includeWitness:
+		case tx.MsgTx().HasWitness() && segwitActive:
+			// If we're about to include a transaction bearing
+			// witness data, then we'll also need to include a
+			// witness commitment in the coinbase transaction.
+			// Therefore, we account for the additional weight
+			// within the block.
+			if !witnessIncluded {
+				// First we account for the additional witness
+				// data in the witness nonce of the coinbaes
+				// transaction: 32-bytes of zeroes.
+				blockWeight += 2 + 32
```

Comment:
What do you think about updating all of this to instead create the coinbase transaction earlier and update it with a fake commitment (all 0s) so it can rely on `wire` to get these sizes?  This seems like it could easily get out of date having all of the magic numbers in here.